app.rb

```diff
@@ -0,0 +1,142 @@
+require_relative './lib/item'
+require_relative './lib/order'
+
+require_relative './lib/item_repository'
+require_relative './lib/order_repository'
+require_relative './lib/database_connection'
+require 'date'
+
+class Application
+  attr_accessor :new_item, :new_order, :time
+  def initialize(database_name, io, item_repository, order_repository)
+    DatabaseConnection.connect(database_name)
+    @io = io
+    @item_repository = item_repository
+    @order_repository = order_repository
+  end
+
+  def run
+    welcome_strings
+    case @io.gets.to_i
+    when 1 then list_all_shop_items
+    when 2 then create_a_new_item
+    when 3 then list_all_orders
+    when 4 then create_a_new_order
+    else fail "Choice must be an integer between 1 and 4"
+    end
+  end
+  
+  private
+
+  def welcome_strings
+    @io.puts "Welcome to the shop management program!"
+    @io.puts "\nWhat would you like to do?"
+    @io.puts "  1 = list all shop items"
+    @io.puts "  2 = create a new item"
+    @io.puts "  3 = list all orders"
+    @io.puts "  4 = create a new order"
+  end
+
+  def list_all_shop_items
+    @io.puts "\nHere's a list of all shop items:"
+    @item_repository.all.each do |item|
+      @io.puts "##{item.id} - #{item.name} - £#{item.unit_price} - #{item.quantity} in stock"
+    end
+  end
+
+  def create_a_new_item
+    ask_for_new_item_info
+    @item_repository.create(new_item)
```

Comment:
Interesting that you're using class level variables like ```new_item```. I hadn't thought of this and wonder about the benefits / drawbacks between this approach and if you'd passed in ```new_item``` as an argument?